src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+package mate.academy.bookdao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dbconnection.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public int getRowsCount() {
+        String sql = "SELECT COUNT(*) AS total_rows FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql);
+                        ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (resultSet.next()) {
+                return resultSet.getInt("total_rows");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get row count", e);
+        }
+        return 0;
+    }
+
+    @Override
+    public boolean clear() {
+        String sql = "TRUNCATE TABLE books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        Statement statement = connection.createStatement()) {
+            statement.executeUpdate(sql);
+            return true;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't clear table", e);
+        }
+
+    }
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books(title, price) VALUES(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setObject(1, book.getTitle());
+            preparedStatement.setInt(2, book.getPrice());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book could not be created");
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                long id = generatedKeys.getLong(1);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add a book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql)) {
```

Comment:
Use `PreparedStatement` over `Statement`. Since you're already using `PreparedStatement`, this is just a reminder to maintain this practice.